### PR TITLE
zee: add dependencies for hpc-coding-conventions

### PR DESCRIPTION
**Pull request by tristan0x**
Requires #245

### CODE REVIEW

File: var/spack/repos/builtin/packages/zee/package.py [hunk 1]
```diff
@@ -48,6 +29,10 @@ class Zee(CMakePackage):
             description='Compile examples using PETSc')
     depends_on('cmake@3:', type='build')
     depends_on('pkg-config', type='build')
+    depends_on('py-cmake-format', type='build')
+    depends_on('py-pre-commit', type='build')
+    depends_on('py-pyyaml', type='build')
+    depends_on('python@3:', type='build')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/zee/package.py]: @tristan0x  : I wonder what is best way to add these dependencies. These are more development dependencies and not strictly required for deployment  of a package. Should there be variant for such dependencies?

Comment (tristan0x) [reply var/spack/repos/builtin/packages/zee/package.py]: I am not against using a variant to enable those dependencies. Is there a proper variant name for such purpose in spack already?

Comment (tristan0x) [reply var/spack/repos/builtin/packages/zee/package.py]: I was thinking of `dev`. I quickly grep variant names in repos but didn't either find `dev` or any other proper name

Comment (pramodk) [reply var/spack/repos/builtin/packages/zee/package.py]: as you know, there is `develop` version which typically points to master. we don't expect coding converts to be applied for other released versions? in this case I wonder if we can use `develop`.

Comment (tristan0x) [reply var/spack/repos/builtin/packages/zee/package.py]: That is a very good idea. Thanks

**Comment by pramodk** [comment]
Tests are failing with `[2019-02-26T13:02:21.775Z] ==> Error: Package py-cmake-format not found.
`. Rebasing on develop.

**Review by pramodk — APPROVED** [review]

**Pull request merged**